pkg/internal/apis/certmanager/validation/certificate.go

```diff
@@ -167,18 +167,16 @@ func ValidateDuration(crt *internalcmapi.CertificateSpec, fldPath *field.Path) f
 	el := field.ErrorList{}
 
 	duration := util.DefaultCertDuration(crt.Duration)
-	renewBefore := cmapi.DefaultRenewBefore
-	if crt.RenewBefore != nil {
-		renewBefore = crt.RenewBefore.Duration
-	}
 	if duration < cmapi.MinimumCertificateDuration {
 		el = append(el, field.Invalid(fldPath.Child("duration"), duration, fmt.Sprintf("certificate duration must be greater than %s", cmapi.MinimumCertificateDuration)))
 	}
-	if renewBefore < cmapi.MinimumRenewBefore {
-		el = append(el, field.Invalid(fldPath.Child("renewBefore"), renewBefore, fmt.Sprintf("certificate renewBefore must be greater than %s", cmapi.MinimumRenewBefore)))
+	// If spec.renewBefore is set, check that it is not less than the minimum.
+	if crt.RenewBefore != nil && crt.RenewBefore.Duration < cmapi.MinimumRenewBefore {
+		el = append(el, field.Invalid(fldPath.Child("renewBefore"), crt.RenewBefore.Duration, fmt.Sprintf("certificate renewBefore must be greater than %s", cmapi.MinimumRenewBefore)))
 	}
-	if duration <= renewBefore {
-		el = append(el, field.Invalid(fldPath.Child("renewBefore"), renewBefore, fmt.Sprintf("certificate duration %s must be greater than renewBefore %s", duration, renewBefore)))
+	// If spec.renewBefore is set, it must be less than the duration.
+	if crt.RenewBefore != nil && crt.RenewBefore.Duration >= duration {
+		el = append(el, field.Invalid(fldPath.Child("renewBefore"), crt.RenewBefore.Duration, fmt.Sprintf("certificate duration %s must be greater than renewBefore %s", duration, crt.RenewBefore.Duration)))
 	}
 	return el
 }
```

Comment:
Some random thoughts about this change:
  * Shall we delete the `cmapi.DefaultRenewBefore` constant?
  * If not, shall we update its documentation comment to explain that it is no longer used and will be removed in a future release.
  * Am I right in thining that previously, the validation would have prevented a user creating / updating a `Certificate.Spec{Duration: "1d"}` (without a corresponding `RenewBefore` value) because in that case it was compared with the DefaultRenewBefore value? If so, that is a relaxation of our validation, which I think is ok, but wanted to be sure.

Reply:
Thank you for looking at this!

> If not, shall we update its documentation comment to explain that it is no longer used and will be removed in a future release

I think I will add a deprecation notice

> Am I right in thining that previously, the validation would have prevented a user creating / updating a Certificate.Spec{Duration: "1d"} (without a corresponding RenewBefore value) because in that case it was compared with the DefaultRenewBefore value? If so, that is a relaxation of our validation, which I think is ok, but wanted to be sure.

Yes, that's correct, I updated the description of this PR with this. I also think this was more like a nuisance than something of value